Libraries/Utilities/defineLazyObjectProperty.js

```diff
@@ -35,7 +35,21 @@ function defineLazyObjectProperty<T>(
     // `setValue` which calls `Object.defineProperty` which somehow triggers
     // `getValue` again. Adding `valueSet` breaks this loop.
     if (!valueSet) {
-      setValue(get());
+      try {
+        // Set `valueSet` to true optimistically to not fail into another infinite
+        // loop when property getter would try to access already defined property.
+        // This is specifically the case with 'regenerator-runtime' module, which
+        // checks if `global.regeneratorRuntime` property is already defined and
+        // thus triggers `getValue()` again.
+        valueSet = true;
```

Comment:
Why it's dangerous? The getter is synchronous and nobody can access property's value from outside until `get()` call is completed. It's only possible to access it from 'inside', i.e. recursively, but in this case we'd get an infinite recursion, which we need to stop.